glitch/analysis/security.py

```diff
@@ -1,15 +1,47 @@
+import os
 import re
 import json
 import configparser
 from urllib.parse import urlparse
-from glitch.analysis.rules import Error, RuleVisitor
+from typing import Tuple, List, Optional
+
+import glitch
+from glitch.analysis.rules import Error, RuleVisitor, SmellChecker
 
 from glitch.repr.inter import *
+from glitch.tech import Tech
 
 
 class SecurityVisitor(RuleVisitor):
     __URL_REGEX = r"^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([_\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$"
 
+    class NonOfficialImageSmell(SmellChecker):
+        def check(self, element, file: str) -> List[Error]:
+            return []
+
+    class DockerNonOfficialImageSmell(SmellChecker):
+        def __init__(self, official_images: List[str]):
+            super().__init__()
+            self.off_images = official_images
+
+        def check(self, element, file: str) -> List[Error]:
+            if not isinstance(element, UnitBlock) or \
+                    element.name is None or "Dockerfile" in element.name:
+                return []
+            image = element.name.split(":")
+            if image[0] not in self.off_images:
+                return [Error('sec_non_official_image', element, file, repr(element))]
+            return []
+
+    def __init__(self, tech: Tech) -> None:
+        super().__init__(tech)
+
+        if tech == Tech.docker:
+            self.non_off_img = SecurityVisitor.DockerNonOfficialImageSmell(
+                self._load_data_file('official_docker_images'))
```

Comment:
The "loading" of the official docker images should be changed to the config as well. Just to keep consistency

Reply:
This was done since for some reason if the loading was done in the same manner the Smell checker could not access it, stating that the field does not exist

Reply:
Hmm that may be because of the "__" before the variable name. Those variables are supposed to be private and maybe because of that inner classes can't access them. This is just a theory (I am still learning about these kinds of things in Python), but you should be able to access them since the config initialization runs before the analysis are executed. Can you try this without the "__" (and if it works try only with "_", this means the field is protected -- should work)?

Reply:
The `__` in Python is only a standard, it does not actually protect the field

Reply:
I also thought so, until the moment I realized that is not completely true. Python does send an attribute error when you try to access a field with `__`. It is not only a standard.

![imagem](https://github.com/sr-lab/GLITCH/assets/32967989/41fef76d-c780-4464-9fa1-c88d726b4180)
![imagem](https://github.com/sr-lab/GLITCH/assets/32967989/ed68d2f3-beb6-4bd8-8d55-fceb999297b5)

Reply:
If you want, you can still access the field, but it has this mechanism to prevent mistakes.

Reply:
So it seems that either I was wrong or something has changed in the meantime, the __ will actually throw an Attribute Error